backend/clubs/views.py

```diff
@@ -1201,6 +1202,31 @@ def create(self, request, *args, **kwargs):
                 f"please email {settings.FROM_EMAIL} if this is en error."
             )
 
+        if request.data.get("is_recurring", None) is not None:
+            parent_recurring_event = RecurringEvent.objects.create()
+            event_data = request.data.copy()
+            start_time = parse(event_data.pop("start_time"))
+            end_time = parse(event_data.pop("end_time"))
+            offset = event_data.pop("offset")
+            end_date = parse(event_data.pop("end_date"))
+            event_data.pop("is_recurring")
+            club_code = kwargs["club_code"]
+
+            club = Club.objects.get(code=club_code)
+            while start_time < end_date:
+                Event.objects.create(
+                    **event_data,
```

Comment:
This will actually allow you to bypass the permissions checks that we have set up for activity fair types, which occur in the serializer.

You can replicate all of the checks here, but that will result in some code duplication which is generally not a good idea. A clean solution would be to let the serializer parse the data for you, and use the parsed data without using the serializer to save. See [this post](https://stackoverflow.com/a/46881848) for an example on how to do so.

Reply:
This will also fix the error issue mentioned in the comment above, as long as you return a good message for when the serializer input ins invalid.